iothub/device/src/DeviceClient.cs

```diff
@@ -436,6 +437,7 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// <param name="blobName">The name of the blob to upload.</param>
         /// <param name="source">A stream with blob contents. Should be disposed after upload completes.</param>
         /// <returns>AsncTask</returns>
+        [Obsolete("This API has been split into three APIs: GetFileUploadSasUri, uploading to blob directly using the Azure Storage SDK, and CompleteFileUploadAsync")]
```

Comment:
Also please mention last SDK version where this API works in comments so that current users of this API know when/what to upgrade to.

Reply:
Users of older versions won't see this message, so I don't see the value

Reply:
I think what I want is when I upgrade and if I happened to be using older API which will now show as deprecated, and can still compile. When I hover over it I want to know what should I do next and when was it deprecated so that I can decide to go back if I dont want to use new approach (for whatever reason) vs upgarde to latest for whatever benefits I am getting. If this is not the right place then where is it?

Reply:
When a method is deprecated isn't interesting to anyone, in my opinion. We don't break the deprecated API in question, so there is never a need for the customer to revert their SDK version

Reply:
As for what they should do instead, that is interesting to everyone, so I included it in the Obsolete comment